torch/csrc/utils/python_arg_parser.h

```diff
@@ -769,13 +769,24 @@ static auto check_has_torch_function(PyObject* obj) -> bool
  *
  * 'torch_api' is a reference to a python torch API namespace.
  *
+ * 'torch_api_function' is the reference to the original torch method, usually,
+ * we can use torch_api and func_name to get torch_api_function. In some cases,
+ * e.g., torch custom op, we create the funciton in C++, if we still use
+ * torch_api and func_name to fetch original api, a cyclic call will happen.
+ *
+ * 'overloaded_args' is the args which have overloaded __torch_function__.
+ *
+ * 'func_name' is the named of the original torch method.
  */
 // Used for Tensor methods with arguments.
 auto handle_torch_function(PythonArgs &r, PyObject* self, PyObject* args, PyObject* kwargs, PyObject* torch_api, const char* module_name) -> PyObject*;
 
-// Used fpr functions.
+// Used for functions which needs to parse python args.
 auto handle_torch_function(PythonArgs &r, PyObject* args, PyObject* kwargs, PyObject* torch_api, const char* module_name) -> PyObject*;
 
+// Used for functions created in C++, e.g., C++ custom op, whose args are already parsed.
+auto handle_torch_function(const std::vector<py::handle> &overloaded_args, PyObject* args, PyObject* kwargs, const char* func_name, PyObject* torch_api_function, const char* module_name) -> PyObject*;
```

Comment:
nit: `ArrayRef<py::handle>` is probably better as it doesn't mandate that you specifically use vector here